### PR TITLE
feat(uat): add test cases to T102 for request/response features

### PR DESCRIPTION
**Pull request by bgklika**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-192
Add test cases to T102 for request/response features

**Description of changes:**
- Add steps to set/reset clean session, keep alive, request response information, response topic tx/rx, correlation data tx/rx
- Add test cases for reponse topic, correlation data and request response information to T102 scenario outline
- Add response topic and correlation data to EventFilter
- Fix unhandled exception and tune logging in paho java client

**Why is this change necessary:**
Automatically test steps, clients and local broker with some new MQTT v5.0 features

**How was this change tested:**
Automatically run scenarios on CodeBuild 

**Test results:**
```
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-sdk-java: As a customer, I can use new MQTT v5.0 features'
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-mosquitto-c: As a customer, I can use new MQTT v5.0 features'
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-paho-java: As a customer, I can use new MQTT v5.0 features'
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-paho-python: As a customer, I can use new MQTT v5.0 features'
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public String getConnectionName() {
 
     @Override
     public boolean isMatched(@NonNull EventFilter filter) {
+        return isMatched1(filter) && isMatched2(filter);
+    }
+
+    private boolean isMatched1(@NonNull EventFilter filter) {
```
Comment (auarbekov) [inline uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java]: can we make the naming more explicit?

File: uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java [hunk 2]
```diff
@@ -72,7 +76,11 @@ public boolean isMatched(@NonNull EventFilter filter) {
             }
         }
 
-        matched = isRetainMatched(filter.getRetain());
+        return true;
+    }
+
+    private boolean isMatched2(@NonNull EventFilter filter) {
```
Comment (auarbekov) [inline uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java]: can we make the naming more explicit?

**Review by eschastlivtsev — DISMISSED** [review]

File: uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java (hunk 1 above)
Comment (bgklika) [reply uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java]: fixed in f64644247dba164315d41899ae2826333a5de17a

File: uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java (hunk 2 above)
Comment (bgklika) [reply uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java]: fixed in f64644247dba164315d41899ae2826333a5de17a

**Review by auarbekov — DISMISSED** [review]

**Pull request closed without merging**